src/RingGrids/full_grids.jl

```diff
@@ -77,128 +52,31 @@ function each_index_in_ring!(   rings::Vector{<:UnitRange{<:Integer}},
-around the poles. The values of all grid points are stored in a vector field `v` that unravels
-the data 0 to 360˚, then ring by ring, which are sorted north to south."""
-struct FullGaussianGrid{T} <: AbstractFullGrid{T}
-    data::Vector{T}    # data vector, ring by ring, north to south
-    nlat_half::Int  # number of latitudes on one hemisphere
-
-    FullGaussianGrid{T}(data::AbstractVector, nlat_half::Integer) where T = length(data) == 8nlat_half^2 ?
-    new(data, nlat_half) : error("$(length(data))-element Vector{$(eltype(data))} cannot be used to create a "*
-        "F$nlat_half ($(4nlat_half)x$(2nlat_half)) FullGaussianGrid{$T}.")
-end
-
-nonparametric_type(::Type{<:FullGaussianGrid}) = FullGaussianGrid
-
-npoints_gaussian(nlat_half::Integer) = 8nlat_half^2
-nlat_half_gaussian(npoints::Integer) = round(Int, sqrt(npoints/8))
-
-# infer nlat_half from data vector length, infer parametric type from eltype of data
-FullGaussianGrid{T}(data::AbstractVector) where T = FullGaussianGrid{T}(data, nlat_half_gaussian(length(data)))
-FullGaussianGrid(data::AbstractVector, n::Integer...) = FullGaussianGrid{eltype(data)}(data, n...)
-
-nlat_odd(::Type{<:FullGaussianGrid}) = false
-get_npoints(::Type{<:FullGaussianGrid}, nlat_half::Integer) = npoints_gaussian(nlat_half)
-get_colat(::Type{<:FullGaussianGrid}, nlat_half::Integer) =
-            π .- acos.(FastGaussQuadrature.gausslegendre(2nlat_half)[1])
-get_quadrature_weights(::Type{<:FullGaussianGrid}, nlat_half::Integer) = gaussian_weights(nlat_half)
-full_grid(::Type{<:FullGaussianGrid}) = FullGaussianGrid    # the full grid with same latitudes
-
-"""
-    G = FullHEALPixGrid{T}
-
-A full HEALPix grid is a regular latitude-longitude grid that uses `nlat` latitudes from the HEALPix grid,
-and the same `nlon` longitudes for every latitude ring. The grid points are closer in zonal direction
-around the poles. The values of all grid points are stored in a vector field `v` that unravels
-the data 0 to 360˚, then ring by ring, which are sorted north to south."""
-struct FullHEALPixGrid{T} <: AbstractFullGrid{T}
-    data::Vector{T}     # data vector, ring by ring, north to south
-    nlat_half::Int      # number of latitudes on one hemisphere
-
-    FullHEALPixGrid{T}(data::AbstractVector, nlat_half::Integer) where T = length(data) == npoints_fullhealpix(nlat_half) ?
-    new(data, nlat_half) : error("$(length(data))-element Vector{$(eltype(data))} cannot be used to create a "*
-        "H$nlat_half ($(4nlat_half)x$(2nlat_half-1)) FullHEALPixGrid{$T}.")
+## COORDINATES
+function get_lond(Grid::Type{<:AbstractFullGridArray}, nlat_half::Integer)
+    lon = get_lon(Grid, nlat_half)
+    lon .*= 360/2π      # convert to lond in-place
+    return lon          # = lond
 end
 
-nonparametric_type(::Type{<:FullHEALPixGrid}) = FullHEALPixGrid
+function get_colatlons(Grid::Type{<:AbstractFullGridArray}, nlat_half::Integer)
```

Comment:
I think some of these functions for `full_grids.jl` deserve a docstring, at least those that can be used for all grid types.

Reply:
Yes, absolutely, docstrings will be added